llvm/test/CodeGen/AMDGPU/sdwa-peephole.ll

```diff
@@ -2102,15 +2102,18 @@ define void @crash_lshlrevb16_not_reg_op() {
 ; NOSDWA:       ; %bb.0: ; %bb0
 ; NOSDWA-NEXT:    s_waitcnt vmcnt(0) expcnt(0) lgkmcnt(0)
 ; NOSDWA-NEXT:    s_mov_b64 s[4:5], 0
+; NOSDWA-NEXT:    s_and_b32 s6, s4, 0xff
```

Comment:
This looks like a missed opportunity for const prop (SI-Fold). Previous case shows const prop happening across BB. This was a case where <2 x i8> was not coerced and ended up producing worse code. At first glance, it looks like coercing needed less const prop and was able reach better looking assembly after SI-Fold pass. Wondering if this is the case of not re-running SI-Fold on changed MI's.